src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -980,6 +965,32 @@ private void SkipWhiteSpace()
             }
         }
 
+        private void ScanNewline(char c)
+        {
+            if (TokenList != null)
+            {
+                _tokenStart = _currentIndex - 1;
+                SkipNewline(c);
+                NewToken(TokenKind.NewLine);
+            }
+        }
+
+        private void ScanSemicolon()
```

Comment:
The same. The name confuse me. We do something only if TokenList != null.

Reply:
The TokenList check has been moved slightly, with a comment explaining why it is there. The name stays the same.